### PR TITLE
PHPLIB-644: Unified GridFS tests

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPLIB-644

POC for https://github.com/mongodb/specifications/pull/963

### CODE REVIEW

File: tests/UnifiedSpecTests/Operation.php [hunk 1]
```diff
@@ -471,18 +471,12 @@ private function executeForBucket(Bucket $bucket)
...
                     array_diff_key($args, ['filename' => 1])
                 ));
             case 'download':
                 return stream_get_contents($bucket->openDownloadStream($args['id']));
-            case 'uploadWithId':
-                $args['_id'] = $args['id'];
-                unset($args['id']);
-
-                // Fall through
-
```
Comment (alcaeus) [inline tests/UnifiedSpecTests/Operation.php:485]: The unified test format spec still references this method, can you create a ticket or PR to https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#upload-and-uploadwithid?

**Review by alcaeus — APPROVED** [review]
LGTM. Do you want to remove the legacy GridFS tests `MongoDB\Tests\GridFS\SpecFunctionalTest` in a separate PR or push changes to this one?

File: tests/UnifiedSpecTests/Operation.php (hunk 1 above)
Comment (jmikola) [reply tests/UnifiedSpecTests/Operation.php:485]: Oh, I totally missed that. Will revert this change.

The method should absolutely exist, as it's noted in the [GridFS spec](https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#file-upload) itself.

**Comment by jmikola** [comment]
I'll remove the legacy GridFS tests in a separate commit in this PR.

**Pull request merged**